### ISSUE TITLE
EDITOR: Cannot resolve import paths in some cases

### ISSUE THREAD

**Issue by catloversg** [bitburner-official/bitburner-src#1747]
TypeScript support will be available on the stable version via #1216. In some cases, monaco cannot resolve the import paths. This is just a problem with the script editor UI. It does not affect anything else. It may be fixed by #1477, but I don't think that fix can be merged easily.

MRE:
lib.ts
```js
export const multiply = (a: number, b: number) => a * b;
```

test.ts
```js
import { multiply } from "/lib";

export async function main(ns: NS) {
  ns.print(`multiply: ${multiply(2, 3)}`);
}
```

Screenshot:
![Capture](https://github.com/user-attachments/assets/982395a3-1c20-43b9-ae1a-3dd1cfcff048)

With this case (use an absolute path), the player can work around by using a relative path (`./lib` instead of `/lib`).

I'll use this issue to keep track of the problem and potential fixes. Other reports can be marked "duplicate" and linked back to here.

Test suite:
[test suite for editor.gz](https://github.com/user-attachments/files/18240475/test.suite.for.editor.gz)

**Comment by d0sboots** [comment]
Yes, we should assume #1477 is dead in the water and steal pieces from it that are useful or mergeable.

**Comment by lucebac** [comment]
I've seen this error in the past as well when I updated the import resolution. However, I believe I wasn't able to fix this because from what I understand this is an edge case of the way we interact with and provide model data to monaco.

**Comment by catloversg** [comment]
Update:
The workaround (using a relative path) still has some problems. In #1661, we added `loadAllServerScripts` function to preload models. The new function has some problems. I tried to fix them in #1865 and #1867. However, a player reported another problem in #1877. If we preload too many models, monaco fails to provide type hints via JSDoc. It still works in TS files, though. I'm too lazy to debug the internal of monaco to see why it happens. Based on https://github.com/microsoft/monaco-editor/issues/3895, I conclude that monaco is not supposed to support loading hundreds of models at the same time, so the "preloading all scripts" solution is only a workaround until we find a proper way to load modules on demand. I'll take a look at the idea in #1477.

**Comment by catloversg** [comment]
Update:
The workaround works properly now with the fix in #1893.
